ir/state.cpp

```diff
@@ -82,26 +79,30 @@ bool State::startBB(const BasicBlock &bb) {
   ENSURE(seen_bbs.emplace(&bb).second);
   current_bb = &bb;
 
+  bool isFirstBB = f.getFirstBB().getName() == bb.getName();
```

Comment:
instead of this huge change, can we do instead (since none of the code later is relevant):
```
if (f.getFirstBB().getName() == bb.getName())
  return true;
```